compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -480,22 +480,33 @@ private boolean isAssignable(BType source, BType target, Set<TypePair> unresolve
             return true;
         }
 
-        if (TypeTags.isXMLTypeTag(sourceTag) && targetTag == TypeTags.XML) {
+        if (TypeTags.isXMLTypeTag(sourceTag) && TypeTags.isXMLTypeTag(targetTag)) {
+            return isXMLTypeAssignable(source, target, unresolvedTypes);
+        }
+
+        if (sourceTag == TypeTags.CHAR_STRING && targetTag == TypeTags.STRING) {
```

Comment:
Merge conflict?
L491  is the same

Reply:
Yep, probably due to merge conflicts. Fixed.